hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileInfo.java

```diff
@@ -380,7 +386,8 @@ private HFileContext createHFileContext(Path path,
     HFileContextBuilder builder = new HFileContextBuilder()
       .withHBaseCheckSum(true)
       .withHFileName(path.getName())
-      .withCompression(trailer.getCompressionCodec());
+      .withCompression(trailer.getCompressionCodec())
+      .withCellComparator(trailer.createComparator(trailer.getComparatorClassName()));
```

Comment:
Better to have a trailer.getComparator() which internally do this create?

Reply:
I want to keep context immutable so it is safe to use in any context. It is why I removed the setter methods. I think it fine building a new one here rather than reuse. Its once per file reader open. Seems fine.

Let me add your getCellComparator suggestion.

Reply:
Hmm... Its a static. createComparator is probably right name here...